contrib/epee/include/storages/parserse_base_utils.h

```diff
@@ -124,14 +144,22 @@ namespace misc_utils
       \\  Backslash character
 
       */
-      inline void match_string2(std::string::const_iterator& star_end_string, std::string::const_iterator buf_end, std::string& val)
+      inline void match_string2(std::string::const_iterator& star_end_string, std::string::const_iterator buf_end, backed_string_ref& sval)
```

Comment:
I don't think the interface should be changed at all, there's no benefit. When `take_string()` is called, if an allocation was skipped in here, it is then done externally. And the only case where `take_string()` is not called is for the name field in the epee storage class and that function is _always_ creating a `std::string` from that parameter.

The problem with this interface is that sometimes the return-value owns the memory, other times not. I'm not sure that the `backed_string_ref` design makes this clear. It is easy to spot if you look at  the implementation real quick, so I might be too pessimistic. But since there is never any savings of allocations in current usage, I don't see why the transition is helpful.